pkg/keyspace/keyspace.go

```diff
@@ -555,3 +571,58 @@ func (manager *Manager) allocID() (uint32, error) {
 	}
 	return id32, nil
 }
+
+// PatrolKeyspaceAssignment is used to patrol all keyspaces and assign them to the keyspace groups.
+func (manager *Manager) PatrolKeyspaceAssignment() error {
+	return manager.store.RunInTxn(manager.ctx, func(txn kv.Txn) error {
```

Comment:
The only worry is that the transaction could be too large to accepted by etcd if we have too many keyspaces. Do we need to do it in batches?

Reply:
It looks good for me for now. We can revisit the comment above after we collect some rough data.

Reply:
Based on the recent discussion, it looks like we need to process it in batch now.